server/verify.go

```diff
@@ -140,37 +164,67 @@ func VerifyAttestation(attestation *pb.Attestation, opts VerifyOpts) (*pb.Machin
 	return nil, fmt.Errorf("attestation does not contain a supported quote")
 }
 
+func getInstanceInfo(extensions []pkix.Extension) (*pb.GCEInstanceInfo, error) {
+	var gceInstanceInfoBytes []byte
+	for _, ext := range extensions {
+		if ext.Id.String() == cloudComputeInstanceIdentifierOID {
+			gceInstanceInfoBytes = ext.Value
+			break
+		}
+	}
+
+	// If GCE Instance Info extension is not found.
+	if gceInstanceInfoBytes == nil {
+		return nil, nil
+	}
+
+	parsedInstanceInfo := gceInstanceInfo{}
+	if _, err := asn1.Unmarshal(gceInstanceInfoBytes, &parsedInstanceInfo); err != nil {
+		return nil, fmt.Errorf("failed to parse GCE Instance Information Extension: %w", err)
+	}
+
+	// Check production.
+	if !parsedInstanceInfo.SecurityProperties.IsProduction {
+		return nil, nil
```

Comment:
When we discussed this previously you recommended returning an empty machine state instead of an error. If that doesn't seem fitting anymore it can be changed to an error.

Reply:
Oh that's my bad. You are absolutely correct